crates/stages/src/stages/execution.rs

```diff
@@ -359,6 +425,20 @@ impl<EF: ExecutorFactory, DB: Database> Stage<DB> for ExecutionStage<EF> {
         StageId::Execution
     }
 
+    fn poll_execute_ready(
+        &mut self,
+        cx: &mut Context<'_>,
+        _: ExecInput,
+    ) -> Poll<Result<(), StageError>> {
+        if let Some(handle) = self.exex_manager_handle.as_mut() {
+            if handle.poll_ready(cx) == Poll::Pending {
+                return Poll::Pending
+            }
+        }
+
+        Poll::Ready(Ok(()))
+    }
+
     /// Execute the stage
     fn execute(
```

Comment:
i opted to just use some of the internal provider funcs to unwind. ideally we remove these and move them into actual dedicated traits. they are super super messy